### PR TITLE
vampire: portability fixes

### PR DESCRIPTION
**Pull request by dtzWill**
Fix problems encountered using libc++, clang, and/or musl.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/logic/vampire/fix-apple-cygwin-defines.patch [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+diff --git a/Minisat/utils/System.cc b/Minisat/utils/System.cc
+index 6d450417..cebfb93b 100644
+--- a/Minisat/utils/System.cc
++++ b/Minisat/utils/System.cc
+@@ -85,7 +85,7 @@ double Minisat::memUsed() {
+ double Minisat::memUsedPeak(bool) { return memUsed(); }
+ 
+ 
+-#elif defined(__APPLE__)
++#elif defined(defined(__APPLE__))
```
Comment (gebner) [inline pkgs/applications/science/logic/vampire/fix-apple-cygwin-defines.patch]: Is this `defined(defined(...))` incantation intentional?

**Review by gebner — APPROVED** [review]

**Comment by dtzWill** [comment]
Eeep! Nope, dishonor on me and my family. This is wrong, silly sed. Thanks for catching, will update shortly.

Kinda surprised that's not an error....

**Comment by gebner** [comment]
Please try to submit the other fixes upstream as well.

Otherwise looks good to me.

**Comment by dtzWill** [comment]
> Please try to submit the other fixes upstream as well.
> 
> Otherwise looks good to me.

Great, will do! Thanks!

Turns out I actually had an old version of the patch here, whoops.  Updated to what was submitted upstream and with the `defined(defined(...))` fix (thanks!).

**Comment by dtzWill** [comment]
All changes submitted upstream, patches are now fetched from corresponding PR's.

Apologies for the rebase'ing but hopefully this is much nicer :).  Thanks for pushing me to submit upstream :+1:.

Also enabled parallel building :grin:.

**Comment by dtzWill** [comment]
Oh, and more complete fpu/fenv fix :).

**Pull request merged**